src/waiters.js

```diff
@@ -1,36 +1,58 @@
 // src/waiters
 
+const request = require('./request');
 const { vmStates } = require('./constants');
-const { getVMState } = require('./methods');
+const { getVMState, isApplicationPublished } = require('./methods');
 
-const maxTries = 5;
-const retryInterval = 15;
+const composeMethod = ({ method, path }) => (body) => request({ path, method, body });
 
-const waitForState = ({ method, methodArgs, numTries=0, targetId, targetName }) => (
+const waitFor = ({ method, methodArgs, maxTries=5, retryInterval=15, targetName, targetId, targetAttribute, expectedValue }) => {
 
-  method(methodArgs).then(({ state }) => {
-    numTries += 1;
+  return new Promise((resolve, reject) => {
```

Comment:
This should probably use the configured promise implementation so that we can have a bluebird promise returned.